src/Surfnet/ServiceProviderDashboard/Infrastructure/DashboardBundle/Twig/ServiceSwitcherExtension.php

```diff
@@ -58,7 +58,7 @@ public function getFunctions()
     public function render(Twig_Environment $environment)
     {
         $token = $this->tokenStorage->getToken();
-        if (!$token) {
+        if (!$token || !$token->hasRole('ROLE_ADMINISTRATOR')) {
```

Comment:
I think there shouldn't be authentication logic in templates. I however don't mind if you don't fix it now.

Reply:
I think there should be and even has to be in order to show/hide functionality from certain roles. This would be quite tedious to fix on a higher level. It is a very interesting thought tho! One to consider more often.